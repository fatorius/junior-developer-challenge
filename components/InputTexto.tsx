import { TextInput, TextInputProps } from "react-native";

import React, { useState } from "react";
import VALUES from "@/VALUES";

const InputTexto: React.FC<TextInputProps> = ({ ...otherProps }) => {
	const [emFoco, setEmFoco] = useState<boolean>(false);

	return (
		<TextInput
			onFocus={() => setEmFoco(true)}
			onBlur={() => setEmFoco(false)}
			placeholderTextColor={VALUES.Tema.corDoTextoEscuro}
			style={[
				{
					fontFamily: VALUES.Fontes.regular,
					fontSize: VALUES.TamanhoFonte.small,
					padding: VALUES.Espacamento * 2,
					backgroundColor: VALUES.Tema.corSecundaria,
					borderRadius: VALUES.Espacamento,
					marginVertical: VALUES.Espacamento,
				},
				emFoco && {
					borderWidth: 3,
					borderColor: VALUES.Tema.corPrimaria,
					shadowOffset: { width: 4, height: VALUES.Espacamento },
					shadowColor: VALUES.Tema.corPrimaria,
					shadowOpacity: 0.2,
					shadowRadius: VALUES.Espacamento,
				},
			]}
			{...otherProps}
		/>
	);
};

export default InputTexto;
