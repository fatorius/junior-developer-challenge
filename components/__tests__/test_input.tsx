import React from "react";
import renderer from "react-test-renderer";
import InputTexto from "../InputTexto";

test("Render: InputTexto", () => {
	const tree = renderer.create(<InputTexto />).toJSON();
	expect(tree).toMatchSnapshot();
});
