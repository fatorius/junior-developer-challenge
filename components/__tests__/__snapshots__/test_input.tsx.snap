// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render: InputTexto 1`] = `
<TextInput
  onBlur={[Function]}
  onFocus={[Function]}
  placeholderTextColor="#626262"
  style={
    [
      {
        "backgroundColor": "#f1f4ff",
        "borderRadius": 9.25,
        "fontFamily": "poppins-regular",
        "fontSize": 22.94694968835727,
        "marginVertical": 9.25,
        "padding": 18.5,
      },
      false,
    ]
  }
/>
`;
